packages/flutter_test/lib/src/matchers.dart

```diff
@@ -286,6 +289,33 @@ Matcher matchesGoldenFile(dynamic key) {
   throw ArgumentError('Unexpected type for golden file: ${key.runtimeType}');
 }
 
+/// Asserts that a [Finder], [Future<ui.Image>], or [ui.Image] matches a
```

Comment:
If desired I can simplify so that this matcher only works for 
ui.Image and Future<ui.Image> and not for Finder.